src/components/tuTable/tuTableStore.ts

```diff
@@ -345,10 +389,59 @@ export class TuTableStore {
 				this.headerCount.value++;
 			}
 		});
+		if (localStorage.getItem(`table-${persistentId}`) !== null) {
+			const persistentColumns = JSON.parse(localStorage.getItem(`table-${persistentId}`));
+			const getNextHeader = (i: number) => {
+				const header = persistentColumns.columns[i].header;
+				let requiredHeader: TuHeaderDefn;
+				for (let j = 0; j < this.table.headers.length; j++) {
+					if (this.table.headers[j].field === header) {
+						requiredHeader = this.table.headers[j];
+						break;
+					}
+				}
+				return requiredHeader;
+			};
+			const prevHeaders: TuHeaderDefn[] = [];
+			for (let i = 0; i < this.table.headers.length; i++)
+				prevHeaders.push(getNextHeader(i));
+			this.table.headers = prevHeaders;
+		}
 	}
 
 	public setColumnVisibility (columnIndex: number, value: boolean) {
-		this.table.headers[columnIndex].hidden = value;
+		for (let i = 0; i < this.table.headers.length; i++) {
+			if (this.table.headers[i].index === columnIndex) {
+				this.table.headers[i].hidden = value;
+				break;
+			}
+		}
+	}
+
+	public reOrderTableColumns (indexOne: number, indexTwo: number, persistentId?: string) {
+		let indexOneFound = false;
+		let indexTwoFound = false;
+		for (let i = 0; i < this.table.headers.length; i++) {
+			if (!indexOneFound) {
+				if (this.table.headers[i].index === indexOne) {
+					indexOne = i;
+					indexOneFound = true;
+				}
+			}
+			if (!indexTwoFound) {
+				if (this.table.headers[i].index === indexTwo) {
+					indexTwo = i;
+					indexTwoFound = true;
+				}
+			}
+		}
```

Comment:
At the end of loop, inside, do `if(indexOneFound && indexTwoFound) break;`. This way you stop looping once both flags are found

Reply:
Okay will do that